src/script/interpreter.cpp

```diff
@@ -2063,55 +2093,69 @@ bool VerifyScript(const CScript &scriptSig, const CScript &scriptPubKey,
             return false;
         }
     }
-    stackCopy = stack;
-    {
-        ScriptExecutionData execdata{scriptPubKey};
-        if (!EvalScript(stack, scriptPubKey, flags, checker, metrics, execdata,
-                        serror)) {
+    if (scriptPubKey.size() > 0 && scriptPubKey[0] == OP_SCRIPTTYPE) {
+        if (!VerifyScriptType(stack, scriptPubKey, flags, checker, metricsOut,
+                              serror)) {
             // serror is set
             return false;
         }
-    }
-    if (stack.empty()) {
-        return set_error(serror, ScriptError::EVAL_FALSE);
-    }
-    if (CastToBool(stack.back()) == false) {
-        return set_error(serror, ScriptError::EVAL_FALSE);
-    }
-
-    // Additional validation for spend-to-script-hash transactions:
-    if (scriptPubKey.IsPayToScriptHash()) {
-        // scriptSig must be literals-only or validation fails
-        if (!scriptSig.IsPushOnly()) {
-            return set_error(serror, ScriptError::SIG_PUSHONLY);
+        if (stack.empty()) {
+            return set_error(serror, ScriptError::EVAL_FALSE);
         }
-
-        // Restore stack.
-        swap(stack, stackCopy);
-
-        // stack cannot be empty here, because if it was the P2SH  HASH <> EQUAL
-        // scriptPubKey would be evaluated with an empty stack and the
-        // EvalScript above would return false.
-        assert(!stack.empty());
-
-        const valtype &pubKeySerialized = stack.back();
-        CScript pubKey2(pubKeySerialized.begin(), pubKeySerialized.end());
-        popstack(stack);
-
+        if (!CastToBool(stack.back())) {
+            return set_error(serror, ScriptError::EVAL_FALSE);
+        }
+    } else {
```

Comment:
This else isn't needed and just causes indentation.

Reply:
Actually we need it cuz after the else we check for cleanstack and sigchecks and setting the metrics.